addons/Broseem/Broseem.lua

```diff
@@ -0,0 +1,187 @@
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Giuliano Riccio BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name     = 'broseem'
+_addon.author   = 'Mojo'
+_addon.version  = '1.20161007'
+_addon.commands = {'broseem'}
+
+require('pack')
+require('lists')
+require('logger')
+require('sets')
+
+files = require('files')
+extdata = require('extdata')
+packets = require('packets')
+
+local item = {}
+local hook = false
+
+local stones = {
+    [0] = 'pellucid',
+    [1] = 'fern',
+    [2] = 'taupe',
+    [3] = 'dark matter',
+}
+
+local paths = {
+    [8] = 'melee',
+    [264] = 'ranged',
+    [520] = 'magic',
+    [776] = 'familiar',
+    [1032] = 'healing',
+    [1290] = 'techniques',
+}
+
+local gear = {
+    [27496] = 'Herculean Boots',
+    [27140] = 'Herculean Gloves',
+    [25642] = 'Herculean Helm',
+    [25842] = 'Herculean Trousers',
+    [25718] = 'Herculean Vest',
+    [27497] = 'Merlinic Crackows',
+    [27141] = 'Merlinic Dastanas',
+    [25643] = 'Merlinic Hood',
+    [25719] = 'Merlinic Jubbah',
+    [25843] = 'Merlinic Shalwar',
+    [25716] = 'Odyssean Chestplate',
+    [25840] = 'Odyssean Cuisses',
+    [27138] = 'Odyssean Gauntlets',
+    [27494] = 'Odyssean Greaves',
+    [25640] = 'Odyssean Helm',
+    [25841] = 'Valorous Hose',
+    [27495] = 'Valorous Greaves',
+    [25717] = 'Valorous Mail',
+    [25641] = 'Valorous Mask',
+    [27139] = 'Valorous Mitts',
+    [25720] = 'Chironic Doublet',
+    [27142] = 'Chironic Gloves',
+    [25644] = 'Chironic Hat',
+    [25844] = 'Chironic Hose',
+    [27498] = 'Chironic Slippers',
+    [21754] = 'Aganoshe',
+    [20677] = 'Colada',
+    [20505] = 'Condemners',
+    [21746] = 'Digirbalag',
+    [21072] = 'Gada',
+    [22054] = 'Grioavolr',
+    [22134] = 'Holliday',
+    [21904] = 'Kanaria',
+    [21804] = 'Obschine',
+    [21854] = 'Reienkyo',
+    [20579] = 'Skinflayer',
+    [22113] = 'Teller',
+    [21021] = 'Umaru',
+    [21686] = 'Zulfiqar',
+}
+
+function make_timestamp()
+    return os.date('%Y:%m:%d %H:%M:%S')
+end
+
+local function save_augments()
+    local fh
+    local path = 'data/%s_augments.csv':format(windower.ffxi.get_player().name)
+    if not files.exists(path) then
+        files.create_path('data')
+        fh = io.open(windower.addon_path .. path, 'w')
+        local header = "timestamp,name,stone,path,augment 1,augment 2,augment 3,augment 4,augment 5\r"
+        fh:write(header)
+        fh:close()
```

Comment:
In addition to @alphaONE2's suggestions here you may as well use the files library to write. You already use it to check if the file exists, you should be able to avoid a few of the common low-level pitfalls (like flushing) if you do so.